tools/quorumscan/cmd/main.go

```diff
@@ -0,0 +1,181 @@
+	if err != nil {
+		return err
+	}
+
+	gethClient, err := geth.NewClient(config.EthClientConfig, gethcommon.Address{}, 0, logger)
+	if err != nil {
+		logger.Error("Cannot create chain.Client", "err", err)
+		return err
+	}
+
+	tx, err := eth.NewReader(logger, gethClient, config.BLSOperatorStateRetrieverAddr, config.EigenDAServiceManagerAddr)
+	if err != nil {
+		log.Fatalln("could not start tcp listener", err)
+	}
+	chainState := eth.NewChainState(tx, gethClient)
+
+	logger.Info("Connecting to subgraph", "url", config.ChainStateConfig.Endpoint)
+	ics := thegraph.MakeIndexedChainState(config.ChainStateConfig, chainState, logger)
+
+	var blockNumber uint
+	if config.BlockNumber != 0 {
+		blockNumber = uint(config.BlockNumber)
+	} else {
+		blockNumber, err = ics.GetCurrentBlockNumber()
+		if err != nil {
+			return fmt.Errorf("failed to fetch current block number - %s", err)
+		}
+	}
+	logger.Info("Using block number", "block", blockNumber)
+
+	operatorState, err := chainState.GetOperatorState(context.Background(), blockNumber, config.QuorumIDs)
+	if err != nil {
+		return fmt.Errorf("failed to fetch operator state - %s", err)
+	}
+	operators, err := ics.GetIndexedOperators(context.Background(), blockNumber)
+	if err != nil {
+		return fmt.Errorf("failed to fetch indexed operators info - %s", err)
+	}
+
+	logger.Info("Queried operator state", "count", len(operators))
+
+	operatorIDs := make([]core.OperatorID, 0, len(operators))
+	for opID := range operators {
+		operatorIDs = append(operatorIDs, opID)
+	}
+	operatorAddresses, err := tx.BatchOperatorIDToAddress(context.Background(), operatorIDs)
+	if err != nil {
+		return err
+	}
+	operatorIdToAddress := make(map[string]string)
+	for i := range operatorAddresses {
+		operatorIdToAddress[operatorIDs[i].Hex()] = strings.ToLower(operatorAddresses[i].Hex())
+	}
+
+	quorumMetrics := quorumscan.QuorumScan(operators, operatorState, logger)
+	displayResults(quorumMetrics, operatorIdToAddress, config.TopN)
+	return nil
+}
+
+func humanizeEth(value *big.Float) string {
+	v, _ := value.Float64()
+	switch {
+	case v >= 1_000_000:
+		return fmt.Sprintf("%.2fM", v/1_000_000)
+	case v >= 1_000:
+		return fmt.Sprintf("%.2fK", v/1_000)
+	default:
+		return fmt.Sprintf("%.2f", v)
+	}
+}
+
+func displayResults(results map[uint8]*quorumscan.QuorumMetrics, operatorIdToAddress map[string]string, topN uint) {
+	weiToEth := new(big.Float).SetFloat64(1e18)
+
+	// Create sorted list of quorums
+	quorums := make([]uint8, 0, len(results))
+	for quorum := range results {
+		quorums = append(quorums, quorum)
+	}
+	sort.Slice(quorums, func(i, j int) bool {
+		return quorums[i] < quorums[j]
+	})
+
+	for _, quorum := range quorums {
+		tw := table.NewWriter()
+		rowAutoMerge := table.RowConfig{AutoMerge: true}
+		operatorHeader := "OPERATOR"
+		if topN > 0 {
+			operatorHeader = "TOP " + strconv.Itoa(int(topN)) + " OPERATORS"
+		}
+		tw.AppendHeader(table.Row{"QUORUM", operatorHeader, "ADDRESS", "STAKE", "STAKE"}, rowAutoMerge)
```

Comment:
maybe "STAKE (ETH)" so its clear what the value is

Reply:
Yea, but that kinda breaks when the quorum is using ETH